### PR TITLE
[Azure Stack] Azure Stack HCI

### PR DESCRIPTION
**Pull request by zhoxing-ms**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Azure Stack HCI

File: src/stack-hci/azext_stack_hci/generated/_help.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+# regenerated.
+# --------------------------------------------------------------------------
+# pylint: disable=too-many-lines
+
+from knack.help_files import helps
+
+
+helps['stack-hci cluster'] = """
+    type: group
+    short-summary: stack-hci cluster
```
Comment (Juliehzl) [inline src/stack-hci/azext_stack_hci/generated/_help.py:17]: Could you refine help message here

File: src/stack-hci/azext_stack_hci/generated/_params.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+cluster_name = CLIArgumentType(options_list=['--name', '-n'], help='The name of the cluster.')
+
+
+def load_arguments(self, _):
+
+    with self.argument_context('stack-hci cluster show') as c:
+        c.argument('cluster_name', cluster_name, id_part='name')
+
+    with self.argument_context('stack-hci cluster create') as c:
+        c.argument('cluster_name', cluster_name)
```
Comment (Juliehzl) [inline src/stack-hci/azext_stack_hci/generated/_params.py:29]: duplicated?

File: src/stack-hci/azext_stack_hci/generated/_params.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        c.argument('tags', tags_type)
```
Comment (Juliehzl) [inline src/stack-hci/azext_stack_hci/generated/_params.py:30]: cli-core could automatically mapping the type for tags

File: src/stack-hci/azext_stack_hci/generated/_params.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        c.argument('location', arg_type=get_location_type(self.cli_ctx),
+                   validator=get_default_location_from_resource_group)
+        c.argument('aad_client_id', help='App ID of cluster AAD identity.')
+        c.argument('aad_tenant_id', help='Tenant ID of cluster AAD identity. Default is the current tenant ID')
+
+    with self.argument_context('stack-hci cluster update') as c:
+        c.argument('cluster_name', cluster_name, id_part='name')
```
Comment (Juliehzl) [inline src/stack-hci/azext_stack_hci/generated/_params.py:37]: duplicated?

File: src/stack-hci/azext_stack_hci/generated/_params.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        c.argument('tags', tags_type)
+
+    with self.argument_context('stack-hci cluster delete') as c:
+        c.argument('cluster_name', cluster_name, id_part='name')
```
Comment (Juliehzl) [inline src/stack-hci/azext_stack_hci/generated/_params.py:41]: same

File: src/stack-hci/azext_stack_hci/generated/_params.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        c.argument('cluster_name', cluster_name)
+        c.argument('tags', tags_type)
+        c.argument('location', arg_type=get_location_type(self.cli_ctx),
+                   validator=get_default_location_from_resource_group)
+        c.argument('aad_client_id', help='App ID of cluster AAD identity.')
+        c.argument('aad_tenant_id', help='Tenant ID of cluster AAD identity. Default is the current tenant ID')
+
+    with self.argument_context('stack-hci cluster update') as c:
+        c.argument('cluster_name', cluster_name, id_part='name')
+        c.argument('tags', tags_type)
```
Comment (Juliehzl) [inline src/stack-hci/azext_stack_hci/generated/_params.py:38]: same

File: src/stack-hci/azext_stack_hci/tests/latest/test_azure_stack_hci_scenario.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+import os
+from azure.cli.testsdk import ScenarioTest
+from .. import try_manual, raise_if
+from azure.cli.testsdk import ResourceGroupPreparer
+
+
+TEST_DIR = os.path.abspath(os.path.join(os.path.abspath(__file__), '..'))
+
+
+@try_manual
```
Comment (Juliehzl) [inline src/stack-hci/azext_stack_hci/tests/latest/test_azure_stack_hci_scenario.py:20]: what is the purpose with this decorator?

File: src/stack-hci/azext_stack_hci/generated/_help.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/stack-hci/azext_stack_hci/generated/_help.py]: Done

File: src/stack-hci/azext_stack_hci/tests/latest/test_azure_stack_hci_scenario.py (hunk 7 above)
Comment (zhoxing-ms) [reply src/stack-hci/azext_stack_hci/tests/latest/test_azure_stack_hci_scenario.py:20]: It is automatically generated by codeGen, I haven't used it yet~

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**